configure.ac

```diff
@@ -11,46 +12,64 @@ define(SECTION_TITLE,
 	echo '   $1   ' | sed -e's/./-/g' >&AS_MESSAGE_FD
 ])
 
+
 dnl works for checking define statements in c
 m4_include([m4/AX_CHECK_DEFINE.m4])
 
-freebsd_src=${FREEBSD_SRC:=/usr/src}
-
-dnl Setting appropriate CFLAGS
-
-CFLAGS="-nostdinc 
--I${freebsd_src}/cddl/lib/libumem 
--I${freebsd_src}/sys/cddl/compat/opensolaris/ 
--I${freebsd_src}/sys/cddl/compat/opensolaris 
--I${freebsd_src}/cddl/compat/opensolaris/include 
--I${freebsd_src}/cddl/compat/opensolaris/lib/libumem 
--I${freebsd_src}/cddl/contrib/opensolaris/lib/libzpool/common 
--I${freebsd_src}/sys/cddl/contrib/opensolaris/common/zfs 
--I${freebsd_src}/sys/cddl/contrib/opensolaris/uts/common/fs/zfs 
--I${freebsd_src}/sys/cddl/contrib/opensolaris/uts/common/sys 
--I${freebsd_src}/cddl/contrib/opensolaris/head 
--I${freebsd_src}/sys/cddl/contrib/opensolaris/uts/common 
--I${freebsd_src}/cddl/contrib/opensolaris/lib/libnvpair 
--I${freebsd_src}/cddl/contrib/opensolaris/lib/libuutil/common 
--I${freebsd_src}/cddl/contrib/opensolaris/lib/libzfs/common 
--I${freebsd_src}/cddl/contrib/opensolaris/lib/libzfs_core/common 
-${CFLAGS} -I/usr/include"
-
-dnl Setting appropriate LDFLAGS
 LDFLAGS="-lzfs -lnvpair -lzfs_core -luutil -lgeom -v"
 
 
-AC_DEFINE(NEED_SOLARIS_BOOLEAN)
-AC_DEFINE(_XPG6)
+dnl We allow setting a zof variable which the users can set to force autoconf to consider the system is zof based
+
+AC_ARG_VAR(zof, AS_HELP_STRING([Force configure to run ZoL(f) checks]))
+AC_ARG_VAR(freebsd_src, AS_HELP_STRING([Provide path for freebsd source tree. This defaults to /usr/src/]))
+
+if [[ -f "/usr/local/include/libzfs/libzfs.h" ]] || [[ "$zof" = true ]]; then
```

Comment:
Lets not hardcode /usr/local anywhere here, please.

Lets use an optional --prefix in configure script, defaulting to /usr/local

There are no guarantees everyone uses /usr/local, plus having it configurable makes it easier to support ZoL.